### PR TITLE
Creation time Partitioner

### PR DESCRIPTION
**Pull request by MaxGekk**

### CODE REVIEW

File: src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        boolean hiveIntegration = hiveIntString != null && hiveIntString.toLowerCase().equals("true");
+        Locale locale = new Locale(localeString);
+        DateTimeZone timeZone = DateTimeZone.forID(timeZoneString);
+        init(pathFormat, locale, timeZone, hiveIntegration);
+    }
+
+    @Override
+    public String encodePartition(SinkRecord sinkRecord) {
+        long timestamp;
+        String fieldName = "creation_time";
```
Comment (DimaD) [inline src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java]: Event time field name is `creation_timestamp`

File: src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            Struct struct = (Struct) value;
+            Object creationTime = struct.get(fieldName);
+            Schema.Type type = valueSchema.field(fieldName).schema().type();
+            switch (type) {
+                case INT32:
+                    Number record = (Number) creationTime;
+                    timestamp = record.longValue();
+                    break;
+                default:
+                    log.error("Type {} is not supported as creation_type.", type.getName());
```
Comment (DimaD) [inline src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java:90]: fieldName is extracted into a variable except this log message

File: src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                        DateTimeZone timeZone, boolean hiveIntegration) {
+        this.formatter = getDateTimeFormatter(pathFormat, timeZone).withLocale(locale);
+        addToPartitionFields(pathFormat, hiveIntegration);
+    }
+
+    private static DateTimeFormatter getDateTimeFormatter(String str, DateTimeZone timeZone) {
+        return DateTimeFormat.forPattern(str).withZone(timeZone);
+    }
+
+    public static long getPartition(long timeGranularityMs, long timestamp, DateTimeZone timeZone) {
```
Comment (DimaD) [inline src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java:48]: Is `timestamp` expected to be in milliseconds or in seconds? `creation_timestamp` in our events is in millliseconds

File: src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java (hunk 1 above)
Comment (MaxGekk) [inline src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java]: Renamed

File: src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java (hunk 3 above)
Comment (MaxGekk) [inline src/main/java/io/confluent/connect/hdfs/partitioner/CreationTimePartitioner.java:48]: In milliseconds. This code was borrowed from TimeBasedPartitioner where
long timestamp = System.currentTimeMillis();

**Pull request merged**